Add support for purging unmanaged config fragments

cosmopetrich opened the pull request:
As suggested by @joshuaspence in #39.

This defaults to disabled to ease migration from previous versions.

I've taken a slightly different approach to #2, in that I've decoupled the parameter names from 'inputs' since conf.d directory could foreseeably be used for outputs and filters. I'm more than happy to change the names etc if you'd prefer different ones. 

Please let me know your thoughts, and thanks for all your work on this module!

Commit:
Add support for purging unmanaged config files

This currently defaults to disabled to ease migration from previous
versions.

yankcrime commented on manifests/config.pp:
```diff
@@ -6,14 +6,24 @@
-    group   => 'telegraf',
-    notify  => Class['::telegraf::service'],
-    require => Class['::telegraf::install'],
+  file {
```
Nice, I was wondering when some Puppet 4 syntax would finally creep in ;)

This is great, but you'll need to update the module's metadata to state this as a requirement.  I'll then call it out as a breaking change in version 2.0.

cosmopetrich replied:
Ah sorry, I didn't even think about that when I was writing it. I'm happy to modify it so that it'll work on older Puppets.

yankcrime replied:
Yes, probably a good idea to maintain compatibility with older versions of Puppet for now.

cosmopetrich replied:
Updated. I've repeated the parameters on each file than using `File {}` to avoid scope-related weirdness, however unlikely.

Commit:
Use Puppet3-compatible resource syntax

This doesn't rely on File {} defaults to avoid potential scope
weirdness, however unlikely.
